packages/opentelemetry/src/spanExporter.ts

```diff
@@ -140,7 +142,10 @@ function maybeSend(spans: ReadableSpan[]): ReadableSpan[] {
       createAndFinishSpanForOtelSpan(child, spans, remaining);
     });
 
-    transactionEvent.spans = spans;
+    transactionEvent.spans =
+      spans.length > MAX_SPAN_COUNT
+        ? spans.sort((a, b) => a.start_timestamp - b.start_timestamp).slice(0, MAX_SPAN_COUNT)
```

Comment:
.sort() is not pure and will mutate the array. I would recommend shallow cloning the array first and making this very explicit.

Reply:
I know, but here we already have a copy of an array that is not used anywhere else, so this is safe and "slightly more efficient". But I can leave a comment to make this clear!

Reply:
I added a comment here (and also in the other place) to be explicit about this! :)